tests/publiccloud/patch_and_reboot.pm

```diff
@@ -26,7 +26,7 @@ sub run {
 
     $args->{my_instance}->retry_ssh_command(cmd => "sudo zypper -n ref", timeout => 240, retry => 6);
     ssh_fully_patch_system($args->{my_instance}->public_ip);
-    $args->{my_instance}->softreboot();
+    $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
```

Comment:
I'd either implement the timeout inside the routine:
```perl
$args{timeout} = get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600);
```

Reply:
This remained unresolved?

Reply:
1. well you approved PR so I decided that it is not so important for you . when you do want to get answer please use "request changes" as Felix did above 
2. not a big deal to solve it just in next PR 
3. I am not sure which routine you actually mean here `softreboot` or `wait_for_ssh` ? 
4. Can you please elaborate why you think that having it inside function is better than this version ?

Reply:
ad 4) because global setting ("variables", `get_var`) should be used at the outermost layer, so that library functions remain as general applicable as possible. If you hypothetically would like to use `softreboot` elsewhere with a even larger timeout, the current solution does not allow that. Pavel's suggestion on the other hand gives this freedom to the function.

Reply:
yeah sure I get that in Pavel's version variable will influence all the places . I just doubt that it is only right decision . For now I know that this timeout is needed in `patch_and_reboot` so I did it there. When and **if** it will be needed elsewhere I will see how better to implement this. Maybe it will be more suitable to have **two** different timeouts ? or maybe it will really make sense to move variable  from current PR down the call-stack ? I have no idea simply because this time haven't come . 
But I don't have hard feeling and ready to move that variable if you really-really think that current state is something not acceptable. I would just would treat this as over engineering  and wanted to express why

Reply:
As it is called `PUBLIC_CLOUD_REBOOT_TIMEOUT` I would move it to softreboot and maybe rename from `REBOOT` to `SOFTREBOOT`. But I have no hard feeling either.